### PR TITLE
Payconiq & Activity payments

### PR DESCRIPTION
**Pull request by JustNireon**
Add activity payments and Payconiq as new payment method.

TO-DO:
- [x] Add ideal to export function

### CODE REVIEW

File: app/controllers/members/payments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#:nodoc:
+class Members::PaymentsController < ApplicationController
+  skip_before_action :authenticate_admin!
+  skip_before_action :authenticate_user!, only: [:confirm_add_funds]
+
+  layout 'members'
+  def index
+    @member = Member.find(current_user.credentials_id)
+
+    # @participants =
```
Comment (Mstiekema) [inline app/controllers/members/payments_controller.rb:10]: Please remove the commented part

File: app/javascript/src/admin/payments.js [hunk 2]
```diff
@@ -31,17 +31,13 @@ $(document).on("ready page:load turbolinks:load", function () {
   });
 
   //Update shown transactions with a start date
```
[the remark this answers is not in the document]
Comment (Mstiekema) [reply app/javascript/src/admin/payments.js:33]: Also, remove the added enters as they are unnecessary

File: app/javascript/src/members/main.js [hunk 3]
```diff
@@ -12,8 +12,7 @@ $(document).on("ready page:load turbolinks:load", function () {
     });
   };
 
-  // Callback handler for menu
```
Comment (Mstiekema) [inline app/javascript/src/members/main.js:15]: Why was this file edited? Please revert

File: app/javascript/src/members/member_payments.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+$(document).on('ready page:load turbolinks:load', function() {
```
Comment (Mstiekema) [inline app/javascript/src/members/member_payments.js:1]: As far as I can see we use 2 instead of 4 spaces as indentation. Please try to keep that the same everywhere.

File: app/views/admin/home/index.html.haml [hunk 5]
```diff
@@ -98,7 +98,35 @@
                   %td.text-center
                     %em
                       = I18n.t('admin.front.no_transactions')
+      %div.col-lg-5
+        .card
+          .card-header
+            %i.fa.fa-fw.fa-shopping-cart
+            /// TODO: Translation
+            Laatste activity transacties vandaag
```
Comment (Mstiekema) [inline app/views/admin/home/index.html.haml:106]: Add translation

File: app/views/admin/home/index.html.haml [hunk 6]
```diff
@@ -98,7 +98,35 @@
...
+                    = link_to "#{transaction.member.name}", transaction.member
+
 
+                  %td.d-none.d-md-table-cell
+                    = transaction.activities.map(&:name).join(', ')
+
+              - if @recentactivities.empty?
+                %tr
+                  %td.text-center
+                    %em geen transacties vandaag
```
Comment (Mstiekema) [inline app/views/admin/home/index.html.haml:129]: Add translation

File: app/views/members/payments/index.html.haml [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+                %td.text-center.d-none.d-sm-block
+                  - if !activity.currency(@member).nil? && activity.currency(@member) > 0
+                    - if activity.participants.where(:member => @member).first.paid
+                      %span.label.label-success= I18n.t('activerecord.attributes.participant.paid.true')
+                    - else
+                      %span.label.label-warning= I18n.t('activerecord.attributes.participant.paid.false')
+            - if @participants.empty?
+              %tr
+                %td.text-center
+                  %em Er zijn geen activiteiten die je moet betalen!
```
Comment (Mstiekema) [inline app/views/members/payments/index.html.haml:50]: Add translation

File: app/views/members/payments/index.html.haml [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+                  %td.d-none.d-sm-table-cell= transaction.products
+                  %td
+                    = number_to_currency(transaction.price, :unit => '€')
+                    - if transaction.price > 0
+                      %span.fa.fa-sort-asc.float-right
+                    - else
+                      %span.fa.fa-sort-desc.float-right
+              %tr.form
+                %td.home_mongoose_form_element
+                  = f.text_field :amount, :placeholder => I18n.t('members.home.index.credit'), :pattern => '^\d+(\.|\,)\d{2}$', 'data-toggle' => 'tooltip', 'data-placement' => 'bottom', :title => 'Er wordt €' + String(@payconiq_transaction_costs) + " #{ I18n.t('members.home.index.transaction_costs')}", :required => ''
```
Comment (Mstiekema) [inline app/views/members/payments/index.html.haml:73]: Add translation

File: config/locales/en.yml [hunk 9]
```diff
@@ -281,6 +313,15 @@ en:
             checkout_emptied: The checkout balance was %{balance}, this has been removed from their account.
             info: "%{name} is deleted. Committee, educations, activity participation, and checkout transactions are anonymous for stats."
             mailchimp_queued: Mailchimp account is automatically removed.
+        payment:
+          attributes:
+            checkout: Mongoose credit
+            fees: iDEAL fee
+          blank: Transactie kan niet leeg zijn
```
Comment (Mstiekema) [inline config/locales/en.yml:320]: This is dutch instead of English

File: config/locales/en.yml [hunk 10]
```diff
@@ -141,6 +142,22 @@ en:
         paid:
           'false': Not paid
           'true': Paid
+      payments:
+        payment-status:
+          failed: Failed
+          in_progress: In progress
+          successful: Succesful
+          title: Betaling status
```
Comment (Mstiekema) [inline config/locales/en.yml:150]: Dutch instead of English

File: config/locales/nl.yml [hunk 11]
```diff
@@ -141,6 +144,22 @@ nl:
...
+          failed: Mislukt
+          in_progress: In uitvoering
+          successful: Geslaagd
+          title: Payment status
+        payment-type:
+          ideal: Ideal
+          payconiq_display: Payconiq Checkout
+          payconiq_online: Payconiq Online
+          pin: Pin
+          title: Payment types
```
Comment (Mstiekema) [inline config/locales/nl.yml:158]: English instead of Dutch

Comment (Mstiekema) [reply config/locales/nl.yml:158]: Same with the stuff below

File: db/schema.rb [hunk 12]
```diff
@@ -131,31 +136,17 @@
     t.index ["member_id", "group_id", "year"], name: "index_group_members_on_member_id_and_group_id_and_year", unique: true
   end
 
-  create_table "groups", id: :integer, force: :cascade do |t|
+  create_table "groups", id: :serial, force: :cascade do |t|
     t.string "name"
     t.integer "category"
-    t.text "comments", size: :medium
```
Comment (Mstiekema) [inline db/schema.rb:137]: Not sure if this migration step went okay, so please check

File: test/models/participants_test.rb [hunk 13]
```diff
@@ -1,77 +0,0 @@
-require "test_helper"
```
Comment (Mstiekema) [inline test/models/participants_test.rb:1]: Why was this file removed?

**Review by Mstiekema — CHANGES_REQUESTED** [review]

File: app/views/admin/activities/partials/_edit.html.haml [hunk 14]
```diff
@@ -153,17 +153,22 @@
           .col-md-4
             = f.check_box :is_masters, checked: @activity.is_masters
             = f.label :is_masters
-        
         .row
           .col-lg-6
             = f.check_box :show_participants, checked: @activity.show_participants
             = f.label :show_participants
-
-      .form-group
```
[the remark this answers is not in the document]
Comment (JustNireon) [reply app/views/admin/activities/partials/_edit.html.haml:162]: formatting reasons you can look at it yourself when testing

File: db/schema.rb (hunk 12 above)
Comment (JustNireon) [reply db/schema.rb:137]: Looks pretty wack but pretty sure it did

File: app/models/payment.rb [hunk 15]
```diff
@@ -0,0 +1,223 @@
...
+  validates :description, presence: true
+  validates :amount, presence: true, numericality: true
+  validates :status, presence: true
+  validates :payment_type, presence: true
+
+  enum :status => [:failed, :in_progress, :successful]
+  enum :payment_type => [:ideal, :payconiq_online, :payconiq_display, :pin]
+  enum :transaction_type => [:checkout, :activity]
+  belongs_to :member
+  # validates :member, presence: true
```
Comment (Mstiekema) [inline app/models/payment.rb:18]: Remove comment

File: test/models/participants_test.rb (hunk 13 above)
Comment (Mstiekema) [inline test/models/participants_test.rb:1]: Why was this removed?

File: .hooks/pre-commit [hunk 16]
```diff
@@ -1,33 +0,0 @@
-#!/bin/bash
```
Comment (Mstiekema) [inline .hooks/pre-commit:1]: Why was this removed?

**Review by Mstiekema — CHANGES_REQUESTED** [review]

File: test/models/participants_test.rb (hunk 13 above)
Comment (JustNireon) [reply test/models/participants_test.rb:1]: Old model that i used when first developing this feature.

**Review by Mstiekema — APPROVED** [review]
GO GO GO

**Review by Mstiekema — APPROVED** [review]
LETS GOOOOOOOOOOOOOOOOOOOO
